test/plots/barPlotTests.ts

```diff
@@ -773,6 +773,23 @@ describe("Plots", () => {
         assert.include(["visible", "inherit"], label2.style("visibility"), "label 1 is visible");
         svg.remove();
       });
+
+      it("hides labels that are partially cut off in y", () => {
+        yScale.domain([1, 2]);
+        let texts = barPlot.content().selectAll("text");
+
+        assert.strictEqual(texts.size(), 2, "There should be two labels rendered");
+
+        texts.each(function(d, i) {
+          let textBounding = (<Element> this).getBoundingClientRect();
+          let svgBounding = (<Element> barPlot.background().node()).getBoundingClientRect();
+          let isLabelCutOff = (textBounding.top < svgBounding.top && textBounding.bottom > svgBounding.top)
+            || (textBounding.top < svgBounding.bottom && textBounding.bottom > svgBounding.bottom);
+          assert.isTrue(isLabelCutOff, `label ${i} is partially cut off`);
+          assert.strictEqual(d3.select(this).style("visibility"), "hidden", `label ${i} is not visible`);
```

Comment:
Followup comment since I missed the `domain` setter call above which is a significant part of this test.

Now I currently see why these labels would be cut off.  My main concern is that if `data` were to change in the `beforeEach` loop then this test would be guaranteed to fail since some labels would be visible.  Would it be possible to select only selections that are partially cut off?